.github/workflows/e2e.yaml

```diff
@@ -29,56 +30,78 @@ jobs:
             **/*.md
             **/*.html
             hugo/**
+
       - id: which_files
         name: Which files was changed
         run: |
           echo "One or more files has changed."
           echo "List all the files that have changed: ${{ steps.changed-files.outputs.all_changed_files }}"
           echo "What is any changed ${{ steps.changed-files.outputs.any_changed }}"
+
       - id: docs_only_check
         if: steps.changed-files.outputs.any_changed != 'true'
         name: Check for docs-only changes
         run: echo "docs_only=true" >> $GITHUB_OUTPUT
 
-  end-to-end:
+  e2e-tests:
+    name: e2e on kind ${{ matrix.version }}
     runs-on: ubuntu-latest
     needs:
       - docs_only_check
     if: (needs.docs_only_check.outputs.docs_only != 'true')
     env:
       KUBECONFIG: /home/runner/.kube/kind-grafana-operator-e2e
+    strategy:
+      matrix:
+        version:
+          - v1.27.13
+          - v1.28.9
+          - v1.29.4
+          - v1.30.0
```

Comment:
How long will this take to run.
I wonder if it's worth running for each PR? Could we come up with something smart to only trigger it before doing a release?

Reply:
With matrix, on average, it should take as much time as before (with single job), thanks to parallel execution.

As for the other question, I think we should run e2e-tests for every PR, but I'm certainly open to any other approach.

Reply:
Ahh of course, forgot that they will run in parallel. Then I don't see any issues with them running on each PR.